### PR TITLE
Next vlan range supplement

### PR DESCRIPTION
**Pull request by pszulczewski**
Sub-PR for `next-vlan-range`

### CODE REVIEW

File: nautobot/ipam/forms.py [hunk 1]
```diff
@@ -722,6 +722,8 @@ class IPAddressFilterForm(NautobotFilterForm, TenancyFilterForm, StatusModelFilt
 
 
 class VLANGroupForm(LocatableModelFormMixin, NautobotModelForm):
+    range = forms.CharField(required=False, help_text="Vlan range, default 1-4094 if left blank.")
```
Comment (mzbroch) [inline nautobot/ipam/forms.py:725]: Can it work for more than 255 Chars ?

File: nautobot/core/models/utils.py [hunk 2]
```diff
@@ -265,6 +265,13 @@ def construct_natural_slug(values, pk=None):
 def compress_range(iterable):
     """
     Generates compressed range from an expanded range.
+    For example:
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 100, 101, 102, 103, 104, 105, 1000, 1100, 1101, 1102, 1103, 1104, 1105, 1106]
+        =>
+        iter1: (1, 10)
+        iter2: (1000, 1000)
+        iter3: (100, 105)
+        iter4: (1100, 1106)
```
Comment (glennmatthews) [inline nautobot/core/models/utils.py:274]: Why this ordering?

Comment (mzbroch) [reply nautobot/core/models/utils.py:274]: It yields these tuples: 1-10, 1000, 100-105, 1100-1106 ?

Comment (mzbroch) [reply nautobot/core/models/utils.py:274]: Do You mean why input iterable is sorted ?

File: nautobot/core/tests/test_utils.py [hunk 3]
```diff
@@ -352,6 +352,89 @@ def test_pretty_print_query(self):
                 self.assertEqual(models_utils.pretty_print_query(query), expected)
 
 
+class CompressRangeTest(TestCase):
+    """Tests for compress_range()."""
+
+    def test_compress_range_sparse(self):
+        values = [10, 200, 1500, 2222, 3000, 4096]
```
Comment (mzbroch) [inline nautobot/core/tests/test_utils.py:359]: This can be un-ordered

File: nautobot/ipam/tests/test_models.py [hunk 4]
```diff
@@ -1317,6 +1317,45 @@ def test_get_next_available_vid(self):
         VLAN.objects.bulk_create((VLAN(name="VLAN 4", vid=4, vlan_group=vlangroup, status=status),))
         self.assertEqual(vlangroup.get_next_available_vid(), 6)
 
+        # Next out of range.
+        VLAN.objects.bulk_create((VLAN(name="VLAN 6", vid=6, vlan_group=vlangroup, status=status),))
+        self.assertEqual(vlangroup.get_next_available_vid(), None)
+
+    def test_reassign_range(self):
```
Comment (mzbroch) [inline nautobot/ipam/tests/test_models.py:1324]: `test_range_resize` ?

File: nautobot/core/models/utils.py (hunk 2 above)
Comment (glennmatthews) [reply nautobot/core/models/utils.py:274]: Why is (1000, 1000) before (100, 105)?

Comment (mzbroch) [reply nautobot/core/models/utils.py:274]: Must be a typo. Sorry I didn't see this one.

Comment (mzbroch) [reply nautobot/core/models/utils.py]: Fixed

**Pull request merged**